scene/2d/box2d_world.h

```diff
@@ -244,15 +230,41 @@ class Box2DWorld : public Node2D, public virtual b2DestructionListener, public v
 		virtual float ReportFixture(b2Fixture *fixture, const b2Vec2 &point, const b2Vec2 &normal, float fraction) override;
 	};
 
+	template <class P, void (Box2DCollisionObject::*on_thing_inout)(P *)>
+	class CollisionUpdateQueue {
+	private:
+		struct CollisionUpdatePair {
+			Box2DCollisionObject *function_owner;
+			P *transient;
+		};
+		std::deque<CollisionUpdatePair> queue{};
+
+	public:
+		inline void enqueue(Box2DCollisionObject *p_caller, P *p_transient) {
+			queue.push_back({ p_caller, p_transient });
+		}
+
+		inline void call_and_clear() {
+			while (!queue.empty()) {
+				CollisionUpdatePair *pair = &queue.front();
+				(pair->function_owner->*on_thing_inout)(pair->transient);
+				queue.pop_front();
+			}
+		}
+	};
+
 private:
 	Vector2 gravity;
 	bool auto_step{true};
 	b2World *world = NULL;
 
-	std::list<GodotSignalCaller> collision_callback_queue{};
```

Comment:
My only thought with a generic callback queue, is to use this for other callbacks in the future.

Say in a callback, the user try's to use or call a function (or add or destroy a body) that's not allowed because b2dWorld.IsLocked() == true.   We can intercept those things, and queue them up for processing after the world is done ticking.  Say, someone calls set_transform which right now just maps directory to b2body->SetTransform(), but if the call that when the world is locked, b2d will assert in debug, and just noop in a release build.

I'm beginning to think we should proxy this state, with a dirty bit, and sync stuff after the world is done it's step.   Just future thoughts.